runtime/interpreter/decode.go

```diff
@@ -712,8 +740,12 @@ func (d StorableDecoder) decodeSome() (SomeStorable, error) {
 	}, nil
 }
 
-func checkEncodedAddressLength(addressBytes []byte) error {
-	actualLength := len(addressBytes)
+func checkEncodedAddressLength(decoder *cbor.StreamDecoder) error {
+	actualLength, err := decoder.NextSize()
+	if err != nil {
+		return err
+	}
```

Comment:
If it doesn't then it may need to be extended. We test for Address length so I assume it's possible for the length to be arbitrarily large. We can't meter effectively without knowing the length ahead of time.